rfcs/OperationExpressions.md

```diff
@@ -0,0 +1,391 @@
+# RFC: Operation Expressions
+
+(WORKING TITLE!)
+
+**Proposed by:** [Benjie Gillam](https://twitter.com/benjie) - Graphile
+
+In the [Schema Coordinates RFC](./SchemaCoordinates.md) Mark introduced the
+concept of "schema coordinates" which give a standard human- and
+machine-readable way to unambiguously refer to entities within a GraphQL schema:
+types, fields, field arguments, enum values, directives and directive arguments.
+The scope of that RFC is deliberately very tight, and it serves that goal well,
+providing a one-to-one mapping between the schema coordinates and the schema
+entities.
+
+This RFC is to gather feedback on expansions of the Schema Coordinate syntax
+that could be used for different purposes whilst maintaining familiarity.
+
+## Aim
+
+The aim of this RFC is to give the GraphQL community a standard syntax that
+people, tools and documentation can use to concisely and consistently reference
+GraphQL operation concepts such as paths that is more fluid, expressive, and
+contains more context than the Schema Coordinates RFC that this RFC builds on
+top of.
+
+This is not intended to be a replacement of the Schema Coordinates RFC, but an
+extension to it for a number of additional use-cases.
+
+## Use cases
+
+#### Referencing a position within a GraphQL Operation Document
+
+Imagine you have the following GraphQL query:
+
+```graphql
+{
+  businesses: searchBusinesses(name: "Automotive") {
+    id
+    name
+    owner: personByOwnerId {
+      id
+      name
+      email # <<< HERE
+    }
+  }
+}
+```
+
+You might reference the marked (`<<< HERE`) field with an expression such as:
+
+- `Person.email` - this is the "schema coordinate" which uniquely identifies the
+  field, but lacks context on how we retrieved it
+- `>businesses>owner>email` - given the GraphQL query document, this is
+  sufficient to uniquely identify this specific reference (caveat: duplicate
+  fields would all be referenced with the same expression)
+- `>businesses:searchBusinesses>owner:personByOwnerId>email` - this contains
+  more context than the above, indicating not just the aliases but the actual
+  field names too; with this access to the operation document is not required to
+  determine what was requested
+- `>businesses:searchBusinesses(name:)>owner:personByOwnerId>email` - this
+  contains even more context (the argument names that were used)
+
+These are all valid operation expressions, but they each convey different levels
+of context.
```

Comment:
I really like this - i could imagine being able to specify different levels of verbosity to tooling that generates these